node/src/signature_mngr.rs

```diff
@@ -257,6 +257,10 @@ impl Actor for SignatureManager {
 
         self.secp = Some(CryptoEngine::new());
     }
+
+    fn stopped(&mut self, _ctx: &mut Self::Context) {
+        System::current().stop_with_code(1);
```

Comment:
I include a panic in Handler<GetPkh> in line 379 and it doesnt close the main node

Reply:
I am seeing that it depends in which part of the code panic, in the handlers there would not close the main node, but it will do if the panic is in the functions that send the messages to the handlers

Reply:
Maybe it's because we also need to add `System::current().stop_with_code` to `StorageManagerAdapter`?

Reply:
> Maybe it's because we also need to add System::current().stop_with_code to StorageManagerAdapter?

No, that didn't work. But I fixed it by adding an `impl Drop for StorageManager` that detects if the thread is panicking.

Reply:
This happens because the `SignatureManager` is started using `SyncArbiter::start`. When a `SignatureManager` handler panics, the panic propagates to the `SyncArbiter`, and that kills the actor without calling its `stopped` method. The `SignatureManagerAdapter` stores a reference (`Addr<SignatureManager>`) to this `SyncArbiter` and when the `SyncArbiter` dies, this reference is no longer valid. We could fix it by doing something like:

```rust
if !self.addr.connected() {
    // The StorageManager died, restart
    self.addr = SyncArbiter::start(...);
}
```

But I think that is too complex, and the alternative solution using `impl Drop` is more robust.